src/main/java/no/nav/pto/veilarbportefolje/ensligforsorger/EnsligeForsorgereRepository.java

```diff
@@ -0,0 +1,296 @@
+package no.nav.pto.veilarbportefolje.ensligforsorger;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import no.nav.familie.eksterne.kontrakter.arbeidsoppfolging.VedtakOvergangsstønadArbeidsoppfølging;
+import no.nav.pto.veilarbportefolje.ensligforsorger.domain.*;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static no.nav.common.client.utils.CacheUtils.tryCacheFirst;
+import static no.nav.pto.veilarbportefolje.util.DateUtils.fnrToDate;
+import static no.nav.pto.veilarbportefolje.util.DateUtils.toLocalDateOrNull;
+import static no.nav.pto.veilarbportefolje.util.SecureLog.secureLog;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+@Transactional
+public class EnsligeForsorgereRepository {
+    private final JdbcTemplate db;
+
+    @Qualifier("PostgresJdbcReadOnly")
+    private final JdbcTemplate dbReadOnly;
+
+    @Qualifier("PostgresNamedJdbcReadOnly")
+    private final NamedParameterJdbcTemplate namedDb;
+
+    private final Cache<String, Integer> stonadstypeCache = Caffeine.newBuilder()
+            .expireAfterWrite(120, TimeUnit.MINUTES)
+            .maximumSize(10)
+            .build();
+    private final Cache<String, Integer> vedtakresultatCache = Caffeine.newBuilder()
+            .expireAfterWrite(120, TimeUnit.MINUTES)
+            .maximumSize(10)
+            .build();
+
+    private final Cache<String, Integer> periodeTypeCache = Caffeine.newBuilder()
+            .expireAfterWrite(120, TimeUnit.MINUTES)
+            .maximumSize(10)
+            .build();
+
+    private final Cache<String, Integer> vettakAktivitetstypeCache = Caffeine.newBuilder()
+            .expireAfterWrite(120, TimeUnit.MINUTES)
+            .maximumSize(10)
+            .build();
+
+    public void lagreEnsligeForsorgereStonad(VedtakOvergangsstønadArbeidsoppfølging vedtakOvergangsstønadArbeidsoppfølging) {
+        if (vedtakOvergangsstønadArbeidsoppfølging.getStønadstype().toString().equals("OVERGANGSSTØNAD")) {
```

Comment:
`"OVERGANGSSTØNAD"` = candidate for an enum?

Reply:
Yep yep, fixed